curvefs/src/client/inode_cache_manager.cpp

```diff
@@ -350,5 +350,57 @@ void InodeCacheManagerImpl::ReleaseCache(uint64_t parentId) {
     iAttrCache_->Release(parentId);
 }
 
+void InodeCacheManagerImpl::FlushInodeLoop() {
+    LOG(INFO) << "flush thread is start.";
+    while (1) {
+        VLOG(9) << "icache size " << iCache_->Size() << " ,maxCacheSize "
+                << maxCacheSize_;
+        if (iCache_->Size() > maxCacheSize_) {
+            DeleteLruInodeCache();
+        } else {
+            sleeper_.wait_for(std::chrono::seconds(flushPeriodSec_));
+            FlushInodeOnce();
+        }
+        if (isStop_.load()) {
+            LOG(INFO) << "flush thread is stop.";
+            return;
+        }
+    }
+}
+
+void InodeCacheManagerImpl::DeleteLruInodeCache() {
+    std::shared_ptr<InodeWrapper> inodeWrapper;
+    uint64_t inodeId;
+    bool ok = iCache_->GetLast(&inodeId, &inodeWrapper);
+    if (ok) {
+        if (inodeWrapper->IsDirty() || IsDirtyMapExist(inodeId)) {
+            VLOG(9) << "DeleteLruInodeCache sync dirty inode " << inodeId;
+            inodeWrapper->ClearDirty();
+            dirtyMapMutex_.lock();
+            dirtyMap_.erase(inodeId);
+            dirtyMapMutex_.unlock();
+            auto ret = inodeWrapper->Sync(true);
+            if (ret != CURVEFS_ERROR::OK) {
+                LOG(WARNING) << "DeleteLruInodeCache sync inode " << inodeId
```

Comment:
When sync fails, dirty flag should be reset and added back to dirtymap

Reply:
fixed